src/proto/parameters.proto

```diff
@@ -96,45 +153,94 @@ message PassingConfig
+        (bounds).min_double_value = 0.1,
+        (bounds).max_double_value = 20.0
+    ];
+    // The min distance in meters that the receiver can be from the passer when receiving
+    // a pass.
+    required double receiver_ideal_min_distance_meters = 16 [
+        default                   = 2.0,
+        (bounds).min_double_value = 0.1,
+        (bounds).max_double_value = 20.0
+    ];
+    // The minimum open angle (in degrees) for an ideal shot on enemy goal from the
+    // receiver's position. Any angle greater than this is also considered ideal, where as
+    // angles below this are considered less ideal and are given a lower score.
+    required double min_ideal_pass_shoot_goal_open_angle_deg = 17 [
+        default                   = 10.0,
+        (bounds).min_double_value = 1.0,
+        (bounds).max_double_value = 45.0
+    ];
+    // The min score that ratePassShootScore return. Used to limit the cost function
+    // from returning a very low score, causing the rest of the cost functions to be
+    // ignored.
+    required double min_pass_shoot_score = 18
+        [default = 0.5, (bounds).min_double_value = 0.0, (bounds).max_double_value = 1.0];
+
+    /*****  Visualization parameters *****/
+    // Cost function visualization parameters
     required CostVisualizationConfig cost_vis_config = 10;
-    // The max speed at which a pass should be received
-    required double max_receive_speed = 11
-        [default = 2.0, (bounds).min_double_value = 1.0, (bounds).max_double_value = 4.0];
-    required double backwards_pass_distance = 13
-        [default = 0.5, (bounds).min_double_value = 0.0, (bounds).max_double_value = 2.0];
+    // Pass generator visualization parameters
+    required PassGeneratorVisualizationConfig pass_gen_vis_config = 22;
+
+    /*****  Receiver position generation parameters *****/
+    required ReceiverPositionGeneratorConfig receiver_position_generator_config = 23;
+}
+
+message ReceiverPositionGeneratorConfig
+{
+    // A multiplier to incentivize using the same receiver position as the previous
+    // iteration
+    required double previous_best_receiver_position_score_multiplier = 1
+        [default = 1.5, (bounds).min_double_value = 1.0, (bounds).max_double_value = 3.0];
+    // The number of samples to generate initially for each zone to get a sense of the
+    // best zones.
+    required uint32 num_initial_samples_per_zone = 2
+        [default = 5, (bounds).min_int_value = 1, (bounds).max_int_value = 100];
+    // The number of additional samples to generate for the top zones after the initial
+    // samples. Used to get a more accurate max score.
+    required uint32 num_additional_samples_per_top_zone = 3
+        [default = 20, (bounds).min_int_value = 1, (bounds).max_int_value = 100];
+    // The minimum angle (in degrees) between a receiver, the ball, and a previously
+    // selected receiver.
+    required double min_angle_between_receivers_deg = 4 [
```

Comment:
in the future, it'd be nice to have a tscope feature in which descriptions of parameters pop up when you hover

Reply:
I agree. I added this issue a while ago with some proposed solutions: #3039